app/screens/gallery/renderers/video/video_controls/index.tsx

```diff
@@ -0,0 +1,224 @@
+    speed: number;
+    captionsEnabled?: boolean;
+    hasCaptions?: boolean;
+    isFullscreen: boolean;
+    seekSeconds: 0 | 10 | 30;
+    onPlay: () => void;
+    onPause: () => void;
+    onSeek: (time: number) => void;
+    onRewind: () => void;
+    onForward: () => void;
+    onSpeedChange: (rate: number) => void;
+    onFullscreen: () => void;
+    onCaptionsToggle?: () => void;
+    setShowCustomControls: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type Control = 'play' | 'pause' | 'seek' | 'rewind' | 'forward' | 'selectSpeed' | 'closeSpeedMenu' | 'speed' | 'fullscreen' | 'captions';
+
+const persistentControls = new Set<Control>(['pause', 'selectSpeed']);
+
+const styles = StyleSheet.create({
+    container: {
+        position: 'absolute',
+        top: 0,
+        bottom: 0,
+        left: 0,
+        right: 0,
+        width: '100%',
+    },
+    controlsBackground: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0,
+        backgroundColor: 'rgba(0, 0, 0, 0.4)',
+    },
+    controlsArea: {
+        flex: 1,
+        justifyContent: 'space-between',
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0,
+    },
+});
+
+const VideoControls: React.FC<VideoControlsWithSeekProps> = ({
+    paused,
+    currentTime,
+    duration,
+    speed,
+    onPlay,
+    onPause,
+    onSeek,
+    onRewind,
+    onForward,
+    onSpeedChange,
+    onFullscreen,
+    onCaptionsToggle,
+    captionsEnabled,
+    hasCaptions = false,
+    isFullscreen,
+    seekSeconds,
+    visible,
+    setShowCustomControls,
+}) => {
+    const insets = useSafeAreaInsets();
+    const headerHeight = useDefaultHeaderHeight();
+    const {headerAndFooterHidden} = useLightboxSharedValues();
+
+    const [showSpeedMenu, setShowSpeedMenu] = useState(false);
+
+    const isInteractingWithControlsRef = useRef(false);
+    const hideControlsTimeoutRef = useRef<NodeJS.Timeout>();
+
+    const cancelHideControls = useCallback(() => {
+        if (hideControlsTimeoutRef.current) {
+            clearTimeout(hideControlsTimeoutRef.current);
+            hideControlsTimeoutRef.current = undefined;
+        }
+    }, []);
+
+    const scheduleAutoHideControls = useCallback((control?: Control, delay: number = 4000) => {
+        if (!paused || control === 'play') {
+            hideControlsTimeoutRef.current = setTimeout(() => {
+                setShowCustomControls(false);
+                setShowSpeedMenu(false);
+            }, delay);
+        }
+    }, [paused, setShowCustomControls]);
+
+    const handleControlAction = useCallback((control: Control, action?: () => void) => {
+        if (!visible) {
+            return;
+        }
+
+        cancelHideControls();
+        isInteractingWithControlsRef.current = true;
+        action?.();
+
+        setTimeout(() => {
+            isInteractingWithControlsRef.current = false;
+        }, 300);
```

Comment:
We should remove any timeouts on unmount to prevent leaks.

Reply:
very true.. will fix.